data/World/Spirit Temple.json

```diff
@@ -129,6 +153,12 @@
             "Spirit Temple Silver Gauntlets Chest": "
                 (Small_Key_Spirit_Temple, 5) or
                 (has_explosives and Longshot and (Small_Key_Spirit_Temple, 3))",
+            "Spirit Temple Hall After Sun Block Room Pot 1" : "(((Small_Key_Spirit_Temple, 3) and is_adult) or (is_adult and spirit_temple_shortcuts)) or
+                (has_explosives or (Small_Key_Spirit_Temple, 3) or ((Small_Key_Spirit_Temple, 2) and bombchus_in_logic and not entrance_shuffle)) or
+                ((Small_Key_Spirit_Temple, 5) and has_explosives)",
+            "Spirit Temple Hall After Sun Block Room Pot 2" : "(((Small_Key_Spirit_Temple, 3) and is_adult) or (is_adult and spirit_temple_shortcuts)) or
+                (has_explosives or (Small_Key_Spirit_Temple, 3) or ((Small_Key_Spirit_Temple, 2) and bombchus_in_logic and not entrance_shuffle)) or
+                ((Small_Key_Spirit_Temple, 5) and has_explosives)",
```

Comment:
Hall after sun block room pots should both just be 'True'.

It was funny to me that you got the main room pots correct, with just 'True', but despite both ages being able to get from those pots to the pots in the child climb and post sun block rooms without needing any items at all, you checked for all these keys and things.

Reply:
You can't just put "True" because of the bomb for light room, child could spent the one key but not have explosives to pass. So I'm gonna think really hard now about what you've written and whether or not it works.

Your adult route is correct though written in a slightly convoluted way where it checks is_adult two separate times. Your child route is 5 keys and explosives, so doesn't actually check is_child, but child needs explosives to access this room, and letting adult up on 5 keys doesn't matter because he can already get up on 3 keys. Your shared route is normal except it includes an extra check of 3 keys. This check does nothing b/c child must have explosives to be in this room and if adult has 3 keys it doesn't affect anything.

So in summary I believe this actually works.

Reply:
We need items to confirm that child can use explosives if necessary, or failing that, that adult definitely has the means to enter the room. I believe the minimum we could say is therefore:
(Small_Key_Spirit_Temple, 3) or spirit_temple_shortcuts or
has_explosives or ((Small_Key_Spirit_Temple, 2) and bombchus_in_logic and not entrance_shuffle)